### PR TITLE
[#22] [UI] [iOS] As a user, I can see the available surveys on the home screen

### LINKED ISSUE #22: [UI] [iOS] As a user, I can see the available surveys on the home screen
## Why

The user can see a list of the surveys which have not been finished by the users on the home screen. The user can only see one item on the list at a time.

#### Acceptance criteria

Show a series of small indicator dots representing the available item in the order they were opened. A solid dot denotes the current page. Users cannot tap on a specific dot to go to a specific page. 

Show a horizontal collection view page. Each page contains the title and description of the survey.

The screen's background is according to the cover image URL of the current survey onscreen.

The user can navigate to the next/previous item by swiping the page to one side.

A circle button to go to take the survey.

## Design

- [Design of the home screen](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=30%3A46&scaling=scale-down)

### PR DESCRIPTION
**Pull request by nmint8m**
- Close #22

## What happened 👀

Show the list of the surveys which have not been finished by the users on the home screen

## Insight 📝

N/A

### CODE REVIEW

File: ios/ios/Sources/Constants/Dimens.swift [hunk 1]
```diff
@@ -21,9 +21,11 @@ enum Dimens {
 }
 
 enum PreviewDimens {
-
+
```
Comment (markgravity) [inline ios/ios/Sources/Constants/Dimens.swift:24]: It's a weird space here 🙌

File: ios/ios/Sources/Presentation/Modules/Home/HomeContainerView.swift [hunk 2]
```diff
@@ -11,20 +11,25 @@ import SwiftUI
...
-            VStack {
-                let homeHeaderUiModel = HomeHeaderUiModel(
-                    imageUrl: "https://avatars.githubusercontent.com/u/7391673?s=200&v=4",
-                    dateText: "Monday, June 15",
-                    todayText: localized.homeHeaderToday()
-                )
-                HomeHeaderView(homeHeader: homeHeaderUiModel) {
-                    // TODO: Tap profile image
-                    print("Tap profile image")
+        GeometryReader(content: { geometry in
```
Comment (markgravity) [inline ios/ios/Sources/Presentation/Modules/Home/HomeContainerView.swift:14]: Where do we use the `geometry`? 🤔

File: ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+//
+
+import SwiftUI
+
+struct SurveyItemView: View {
+    @State private var currentPage: Int = 0
+
+    var body: some View {
+        ZStack {
+            GeometryReader { geometry in
```
Comment (markgravity) [inline ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift:16]: How about using `.frame(maxWidth: .infinity, maxHeight: .infinity)` here? 🤔

File: ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                                }
+                            )
+                        }
+
+                    }
+                )
+                .padding(
+                    EdgeInsets(
+                        top: 0.0,
+                        leading: 20.0,
```
Comment (thiennguyen0196) [inline ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift:66]: should we reuse `Dimens.large` with this `20.0` 🤔

File: ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                        bottom: 0.0,
+                        trailing: 20.0
```
Comment (thiennguyen0196) [inline ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift:68]: Ditto here

**Review by thiennguyen0196 — COMMENTED** [review]
I just have some comments about the reuse of common dimens, the rest LGTM 👏

File: ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift (hunk 3 above)
Comment (nmint8m) [reply ios/ios/Sources/Presentation/Modules/Home/SurveyListView/SurveyItemView.swift:16]: No, we cannot, we should use `Geometry` wrap around and `.ignoresSafeArea()` to make the background image expanding with full height.

**Review by markgravity — APPROVED** [review]

**Review by thiennguyen0196 — APPROVED** [review]
LGTM now

**Pull request merged**